lib/dynamoid/config/backoff_strategies/exponential_backoff.rb

```diff
@@ -0,0 +1,24 @@
+module Dynamoid
+  module Config
+    module BackoffStrategies
+      # Truncated binary exponential backoff algorithm
+      # See https://en.wikipedia.org/wiki/Exponential_backoff
+      class ExponentialBackoff
+        def self.call(opts)
+          base_backoff = opts[:base_backoff]
+          ceiling = opts[:ceiling]
```

Comment:
Would we want potential defaults or this should always be configured by the user? Feel like providing some base backoff configs so can just out of box say "let's have exponential backoff" without worrying about underlying details is nice.